web/app/scripts/src/services/calculate-percentage.es.js

```diff
@@ -34,15 +37,21 @@ module.service(moduleProperties.componentName, [
         //     }
         // }
         // answered is a number of type number
-        this.countNumberOfAnsweredQuestions = function (answers) {
+        this.countNumberOfAnsweredQuestions = function (answers, qKeys) {
             let answered = 0;
 
+            if (qKeys.length === 0) {
+                return answered;
+            }
+
             for (const groupKey in answers) {
                 const answerGroup = answers[groupKey];
                 for (const answerKey in answerGroup) {
                     const answer = answerGroup[answerKey];
-                    if (angular.isDefined(answer) && answer !== null) {
-                        answered++;
+                    if (qKeys.indexOf(parseInt(answerKey)) !== -1) {
+                        if (answer && answer !== null || answer === 0) {
```

Comment:
Where did the ` || answer === 0` come from? It doesn't look like it was there before?